### DISCUSSION TITLE
Updatable condition

### DISCUSSION THREAD

**Discussion by harifenny** [serenity-is/Serenity#5406] [Help] [answered]
How to customize Updatable attribute? For example I want to Updatable(false) overall property/field if Status==Approved 

Many thanks, 
Hari Setyawan

**Comment by erehtolleh** [comment, marked as answer]
You can do it in the dialog.ts or editordialog.ts files

You can check for the status and then make it readonly.  You can do this in the constructor or afterLoadEntity() 

Serenity.EditorUtils.setReadonly(this.element.find('.editor'), true);

and also something like below to hide or disable buttons.

 this.deleteButton.hide();
 this.applyChangesButton.toggleClass('disabled', true);

I would also check before save in validateBeforeSave() and check it there.  If it's super sensitive that they don't change, then you probably have to mess with the endpoint or repository files.

If someone else does it differently or better, please chime in.

**Reply by edwardch** [reply]
Yip, looks good.

You can put it in the updateInterface() of your xyzDialog.ts.

```ts
protected updateInterface(): void {

	super.updateInterface();
        // Assuming "Status" is a boolean. If not your syntax might look different
	if (this.form.Status.value) {
		this.deleteButton.hide();
		this.applyChangesButton.hide();
		Serenity.EditorUtils.setReadOnly(this.form.Field1, true);
		Serenity.EditorUtils.setReadOnly(this.form.Field2, true);
	}
}
```

See [CustomerOrderDialog.ts](https://github.com/serenity-is/Serene/blob/master/Serene/Serene.Core/Modules/Northwind/Customer/CustomerOrderDialog.ts) and [EntityDialogAsPanel.ts](https://github.com/serenity-is/Serene/blob/master/Serene/Serene.Core/Modules/BasicSamples/Dialogs/EntityDialogAsPanel/EntityDialogAsPanel.ts) from the Serene demo.

And yes best practice is to also validate server-side in your xyzRepository.cs like so
```cs
private class MySaveHandler : SaveRequestHandler<MyRow>
{
    protected override void ValidateRequest()
    {
        base.ValidateRequest();
        // I haven't tested this, but should be about right. Otherwise you must add your own logic here
        if (!Old.Status.Value && Row.IsAnyFieldChanged)
            throw new ValidationError("Can't update an approved record.");
    }
}
```

HTH

**Comment by erehtolleh** [comment]
There is a [sample](https://serenity.is/demo/BasicSamples/ReadOnlyDialog)

**Reply by harifenny** [reply]
Thank you
